### PR TITLE
Alternate mrw proxy

### PR DESCRIPTION
**Pull request by m-nash**
Alternate implementation of https://github.com/Azure/azure-sdk-for-net/pull/45400 for comparison

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/core/System.ClientModel/samples/ModelReaderWriter.md [hunk 1]
```diff
@@ -53,4 +53,35 @@ string json = @"{
       ""z"": 3
     }";
 OutputModel? model = JsonSerializer.Deserialize<OutputModel>(json, options);
+```
+
+## Using a Proxy
+
+In more advanced scenarios a library user might want to override the behavior of how a model is read or written.
+In this case you can implement your own class which implements the same `IPersistableModel<T>` or `IJsonModel<T>` and register it with the `ModelReaderWriterOptions`.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/samples/ModelReaderWriter.md:61]: > the same `IPersistableModel<T>` or `IJsonModel<T>`

this is a little unclear -- would it be possible ground this in an example?  e.g. something like -- "for example, to add a proxy for a model `InputModel` that implements `IJsonModel<InputModel>`, you would add a type such as `InputModelProxy` that also implements `IJsonModel<InputModel>`.  This clarifies a bit what "the same `IJsonModel<T>` means.

File: sdk/core/System.ClientModel/samples/ModelReaderWriter.md [hunk 2]
```diff
@@ -53,4 +53,35 @@ string json = @"{
...
+
+The example below shows how to write a persitable model using a proxy to `BinaryData`
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/samples/ModelReaderWriter.md:63]: Did you mean "read" here?

```suggestion
The example below shows how to read a persistable model using a proxy to `BinaryData`
```

File: sdk/core/System.ClientModel/samples/ModelReaderWriter.md [hunk 3]
```diff
@@ -53,4 +53,35 @@ string json = @"{
...
+      ""z"": 3
+    }";
+
+ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+options.AddProxy(new OutputModelProxy());
+
+OutputModel? model = ModelReaderWriter.Read<OutputModel>(BinaryData.FromString(json), options);
+```
+
+The example below shows how to read JSON to create a strongly-typed model instance using a proxy.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/samples/ModelReaderWriter.md:78]: Did you mean "write" here?

```suggestion
The example below shows how to write JSON to create request content from a strongly-typed model using a proxy.
```

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 4]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
     /// Gets the format to read and write the model.
     /// </summary>
     public string Format { get; }
+
+    /// <summary>
+    /// Registers an <see cref="IPersistableModel{T}"/> proxy to be used when reading or writing a model.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:43]: nit for clarity:

```suggestion
    /// Registers an <see cref="IPersistableModel{T}"/> proxy model that the ModelReaderWriter or model graph will use reading or writing a model of type <code>T</code>.
```

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 5]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+    public void AddProxy<T>(IPersistableModel<T> proxy)
+    {
+        if (_proxies is null)
+            _proxies = new Dictionary<Type, object>();
+
+        _proxies.Add(typeof(T), proxy);
+    }
+
+    /// <summary>
+    /// Gets the model that is currently being proxied.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:55]: Consider the following clarification:

```suggestion
    /// Gets the instance of a model that is currently being read or written via a proxy at that stage of reading or writing the model graph.
```

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 6]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+    /// </summary>
+    /// <remarks>
+    /// Whenever a proxy is used for a given type, the model being proxied is set here.
+    /// The value does not need to be cleared after setting it will be overwritten on the next use.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:59]: Does this line (59) need to be present if this property is no longer settable by the end-user?

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 7]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+    /// </remarks>
+    public object? ProxiedModel { get; private set; }
+
+    /// <summary>
+    /// Gets the <see cref="IPersistableModel{T}"/> proxy for the specified <typeparamref name="T"/> model type.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:64]: Consider 

```suggestion
    /// Gets a <see cref="IPersistableModel{T}"/> proxy model for the specified <typeparamref name="T"/> model type, if one has been added to this options instance.
```

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 8]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+    /// </summary>
+    /// <param name="proxy"> The <see cref="IPersistableModel{T}"/> proxy if one exists. </param>
+    /// <returns> True if a proxy was found; otherwise, false. </returns>
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:67]:
```suggestion
    /// <returns> True if a proxy for a model of type <code>T</code> was found; otherwise, false. </returns>
```

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 9]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+        {
+            proxy = default;
+            return false;
+        }
+
+        proxy = jsonResult;
+        return true;
+    }
+
+    internal bool TryGetProxy(Type modelType, [NotNullWhen(true)] out IJsonModel<object>? proxy)
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:97]: Do these need to be internal, or could they be private?

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 10]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+            proxy = default;
+            return false;
+        }
+
+        proxy = (IPersistableModel<object>)result;
+        return true;
+    }
+
+    /// <summary>
+    /// Gets the <see cref="IPersistableModel{T}"/> proxy for the specified <typeparamref name="T"/> model type.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:122]: It would be good to add an explanation of what "resolve" means and how it works either here or in `<remarks>` in these docs, as this piece is a little non-standard for .NET patterns.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 11]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+    /// </summary>
+    /// <param name="model"> The <see cref="IPersistableModel{T}"/> model to proxy. </param>
+    /// <returns> The <see cref="IPersistableModel{T}"/> proxy to use. </returns>
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:125]: Maybe...

```suggestion
    /// <returns> The <see cref="IPersistableModel{T}"/> proxy or model to use for reading or writing a model of type <code>T</code>. </returns>
```

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 12]
```diff
@@ -33,4 +38,115 @@ public ModelReaderWriterOptions (string format)
...
+    public IPersistableModel<T> ResolveProxy<T>(IPersistableModel<T> model)
+    {
+        if (_proxies is null || !_proxies.TryGetValue(model.GetType(), out object? result))
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:128]: It might be helpful for future readers/maintainers of this code to add a comment that explains that this checks whether a proxy has been registered for type T, and maybe also that this implements an exact type look-up, and not a proxy for subtypes of T.

File: sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs [hunk 13]
```diff
@@ -83,5 +83,75 @@ public void ValidateCapitalizationIsIgnored()
...
+            Assert.AreEqual(model.RegistrationPolicy, model2.RegistrationPolicy);
+            Assert.AreEqual(model.RegistrationState, model2.RegistrationState);
+            Assert.AreEqual(model.ResourceTypes.Count, model2.ResourceTypes.Count);
+        }
+
+        [Test]
+        public void ProxySerializationOfNestedType()
+        {
+            var options = new ModelReaderWriterOptions("J");
+            options.AddProxy(new ProviderResourceTypeProxy());
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs:140]: It might be helpful to future readers/maintainers of this code to add a comment that ProviderResourceType is a nested type in the ResourceProviderData object graph -- i.e. the type of a property on ResourceProviderData or one of its child types.

File: sdk/core/System.ClientModel/tests/Samples/ModelReaderWriterSamples.cs [hunk 14]
```diff
@@ -58,6 +58,90 @@ public void Stj_Read_Simple()
...
+            }";
+
+        ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+        options.AddProxy(new OutputModelProxy());
+
+        OutputModel? model = ModelReaderWriter.Read<OutputModel>(BinaryData.FromString(json), options);
+        #endregion
+    }
+
+    private class OutputModelProxy : IJsonModel<OutputModel>
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/Samples/ModelReaderWriterSamples.cs:89]: Consider adding this class declaration somewhere in the sample -- it was very helpful for me to see this to understand how proxies actually work.

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ProviderResourceTypeProxy.cs [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+// <auto-generated/>
+
+using ClientModel.Tests.ClientShared;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Text.Json;
+
+namespace System.ClientModel.Tests.Client.Models.ResourceManager.Resources
+{
+    public partial class ProviderResourceTypeProxy : IJsonModel<ProviderResourceType>
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ProviderResourceTypeProxy.cs:13]: A comment here explaining what the class is used for might help future readers/maintainers of this code.

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs [hunk 16]
```diff
@@ -61,7 +62,15 @@ public static ResourceProviderData DeserializeResourceProviderData(JsonElement e
                     List<ProviderResourceType> array = new List<ProviderResourceType>();
                     foreach (var item in property.Value.EnumerateArray())
                     {
-                        array.Add(ProviderResourceType.DeserializeProviderResourceType(item, options));
+                        if (options.TryGetProxy<ProviderResourceType>(out IJsonModel<ProviderResourceType> proxy))
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs:65]: Should other model implementations in the test suite be updated to test for proxies in their implementations?

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ResourceDataWriteProxy.cs [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+
+using System.ClientModel.Primitives;
+using System.ClientModel.Tests.Client.Models.ResourceManager.Resources;
+using System.Collections.Generic;
+using System.Text.Json;
+using ClientModel.Tests.ClientShared;
+
+namespace System.ClientModel.Tests.Client
+{
+    public class ResourceDataWriteProxy : IJsonModel<ResourceProviderData>
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ResourceDataWriteProxy.cs:12]: Curious -- it's called WriteProxy ... is it only for writing or could it be used for reading as well?

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 9 above)
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:97]: they are used by other types in the library so internal

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 12 above)
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:128]: I think the line of code is clear that it is checking if a proxy exists, but added a line at the top of the class stating all proxies are exact matches. since this is the case for all methods not just this one.

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs (hunk 16 above)
Comment (m-nash) [reply sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs:65]: hand writing this is a bit much to maintain we have a full set of autogenerated models in the generator since that is technically the code that owns the implementation.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 7 above)
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:64]: I think `the` is more appropriate here as there can only be one and `a` implies its one of N.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 5 above)
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:55]: I think this makes it more confusing.  Currently being proxied sounds clearer to me.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 4 above)
Comment (m-nash) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:43]: I think its clear that since this is the ModelReaderWriterOptions the ModelReaderWriter will use it, I am not sure what you mean by "model graph" will use, this is not a concept in the MRW apis.

File: sdk/core/System.ClientModel/samples/ModelReaderWriter.md [hunk 18]
```diff
@@ -53,4 +53,47 @@ string json = @"{
       ""z"": 3
     }";
 OutputModel? model = JsonSerializer.Deserialize<OutputModel>(json, options);
+```
+
+## Using a Proxy
+
+In more advanced scenarios a library user might want to override the behavior of how a model is read or written.
+In this case you can implement your own class which implements the same interface, either `IPersistableModel<T>` or `IJsonModel<T>`, and register it with the `ModelReaderWriterOptions`.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/samples/ModelReaderWriter.md:61]: Would it be possible to call out the difference between the new proxy feature and `[PersistableModelProxy]` here?  I expect the reuse of that name to be confusing to customers so it would be nice to provide reference materials to help them understand.

File: sdk/core/System.ClientModel/samples/ModelReaderWriter.md [hunk 19]
```diff
@@ -53,4 +53,47 @@ string json = @"{
...
+      ""x"": 1,
+      ""y"": 2,
+      ""z"": 3
+    }";
+
+ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+options.AddProxy(new OutputModelProxy());
+
+OutputModel? model = ModelReaderWriter.Read<OutputModel>(BinaryData.FromString(json), options);
+```
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/samples/ModelReaderWriter.md:82]: It might be helpful to add a final line here (and below) to illustrate the difference that was achieved by using the proxy instead of the model for reading the content.  What happened that was different?  This will help ground the example for readers.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs [hunk 20]
```diff
@@ -195,7 +223,10 @@ internal static bool ShouldWriteAsJson(IPersistableModel<object> model, ModelRea
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private static bool IsJsonFormatRequested<T>(IPersistableModel<T> model, ModelReaderWriterOptions options)
-        => options.Format == "J" || (options.Format == "W" && model.GetFormatFromOptions(options) == "J");
+    {
+        var format = options.TryGetProxy(out IPersistableModel<T>? proxy) ? proxy.GetFormatFromOptions(options) : model.GetFormatFromOptions(options);
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs:227]: Curious: can the proxy specify a different format from the base model?  It might be helpful to draw out that scenario, maybe in tests?

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs [hunk 21]
```diff
@@ -1,14 +1,21 @@
...
+using System.Diagnostics.CodeAnalysis;
+
 namespace System.ClientModel.Primitives;
 
 /// <summary>
 /// Provides the client options for reading and writing models.
 /// </summary>
 public class ModelReaderWriterOptions
 {
+    // Proxies are compared by exact type only.
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:14]: Would it be possible to add another sentence to this comment describing the implications of this?  i.e. "this means that TryGetProxy will not find a proxy if it has been registered by a base type or derived type."  I think that would help maintainers understand the thrust of this comment better.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 4 above)
Comment (annelo-msft) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:45]: But it will not be used in 100% of cases, correct?  For example, if a caller calls model.Write, it will not use a proxy?  

Given this I think it would be more clear to specify when it will be used in the public refdoc comment.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 5 above)
Comment (annelo-msft) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:57]: The words in the comment could imply that you return the model type, not the model instance.  It would be helpful to users to disambiguate this.

File: sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs (hunk 7 above)
Comment (annelo-msft) [reply sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs:62]: That is fair about `the` and `a`.  It might still be helpful to clarify that if no proxy has been registered, no proxy will be returned.

File: sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelTests.cs [hunk 22]
```diff
@@ -130,7 +130,7 @@ public void ThrowsIfUnknownFormat()
             Assert.Throws<FormatException>(() => ModelReaderWriter.Write(ModelInstance, options));
             Assert.Throws<FormatException>(() => ModelReaderWriter.Read<T>(new BinaryData("x"), options));
 
-            Assert.Throws<FormatException>(() => ModelReaderWriter.Write((IPersistableModel<object>)ModelInstance, options));
+            Assert.Throws<FormatException>(() => ModelReaderWriter.Write((object)ModelInstance, options));
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelTests.cs:133]: Curious: why is this needed?

File: sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs [hunk 23]
```diff
@@ -83,5 +83,76 @@ public void ValidateCapitalizationIsIgnored()
...
+            var model = ModelReaderWriter.Read<ResourceProviderData>(BinaryData.FromString(WirePayload));
+            Assert.NotNull(model);
+            var binaryData = ModelReaderWriter.Write(model!, options);
+            Assert.AreEqual(File.ReadAllText(TestData.GetLocation("ResourceProviderData/ResourceProviderData-Collapsed-MissingId.json")).TrimEnd(), binaryData.ToString());
+            Assert.AreNotEqual(WirePayload, binaryData.ToString());
+
+            var model2 = ModelReaderWriter.Read<ResourceProviderData>(binaryData, options);
+            Assert.NotNull(model2);
+            Assert.NotNull(model2!.Id);
+            Assert.AreEqual("TestValue", model2.Id);
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs:102]: It might be helpful to add a comment here to say that if the model was written without the proxy, this value would be different ... or whatever is the correct way of interpreting these tests so that maintainers can understand the intended behavior of the type.

File: sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs [hunk 24]
```diff
@@ -83,5 +83,76 @@ public void ValidateCapitalizationIsIgnored()
...
+            Assert.AreEqual(File.ReadAllText(TestData.GetLocation("ResourceProviderData/ResourceProviderData-Collapsed-With-x.json")).TrimEnd(), binaryData.ToString());
+            Assert.AreNotEqual(WirePayload, binaryData.ToString());
+
+            var model2 = ModelReaderWriter.Read<ResourceProviderData>(binaryData, options);
+            Assert.NotNull(model2);
+            Assert.AreEqual(model!.Id, model2!.Id);
+            Assert.AreEqual(model.Namespace, model2.Namespace);
+            Assert.AreEqual(model.RegistrationPolicy, model2.RegistrationPolicy);
+            Assert.AreEqual(model.RegistrationState, model2.RegistrationState);
+            Assert.AreEqual(model.ResourceTypes.Count, model2.ResourceTypes.Count);
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ResourceProviderDataTests.cs:155]: Would you add a comment here to help make it clear where the difference is due to the proxy that the test is validating?

File: sdk/core/System.ClientModel/tests/client/ClientShared/ModelReaderWriterExtensions.cs [hunk 25]
```diff
@@ -176,15 +176,16 @@ public static void WriteNumberValue(this Utf8JsonWriter writer, DateTimeOffset v
         writer.WriteNumberValue(value.ToUnixTimeSeconds());
     }
 
-    public static void WriteObjectValue(this Utf8JsonWriter writer, object? value)
+    public static void WriteObjectValue(this Utf8JsonWriter writer, object? value, ModelReaderWriterOptions? options = null)
```
Comment (annelo-msft) [inline sdk/core/System.ClientModel/tests/client/ClientShared/ModelReaderWriterExtensions.cs:179]: Curious: is this a change that will be needed in the generator to write the ModelReaderWriterExtensions file differently to support the proxy feature?

File: sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs (hunk 16 above)
Comment (annelo-msft) [reply sdk/core/System.ClientModel/tests/client/ModelReaderWriter/ServiceModels/ResourceProviderData.Serialization.cs:65]: Will other tests fail in the future because the test client code is not up to date?
